dm/syncer/data_validator.go

```diff
@@ -328,20 +379,57 @@ func (v *DataValidator) doValidate() {
 		}
 	}
 
-	v.L.Info("start continuous validation")
 	v.startValidateWorkers()
 	defer func() {
 		for _, worker := range v.workers {
 			worker.close()
 		}
 	}()
 
+	metaFlushInterval := v.cfg.ValidatorCfg.MetaFlushInterval.Duration
+	// when gtid is enabled:
+	// gtid of currLoc need to be updated when meet gtid event, so we can compare with syncer
+	// gtid of locationForFlush is updated when we meet xid event, if error happened, we start sync from this location
+	//
+	// some events maybe processed again, but it doesn't matter for validation
 	currLoc := location.CloneWithFlavor(v.cfg.Flavor)
+	locationForFlush := currLoc.Clone()
+	lastFlushCheckpointTime := time.Now()
 	for {
 		e, err := v.streamerController.GetEvent(v.tctx)
 		if err != nil {
-			v.errChan <- terror.Annotate(err, "fail to get binlog from stream controller")
-			return
+			switch {
```

Comment:
yes, similar with how syncer.run handles it.

even we merge this switch together with getEvent, we still need a switch kind of thing, since the error returned affect the control path( break/continue/return) of outside code.

maybe we can try to abstract them later when we refactor syncer.